ipa-core/src/ff/boolean_array.rs

```diff
@@ -497,6 +497,28 @@ macro_rules! boolean_array_impl {
                 }
             }
 
+            /// This function converts an `BA` type into a vector of `32 bit` Galois field elements.
+            ///
+            /// ##Errors
+            /// Outputs an error when conversion from raw slice to Galois field element fails.
+            impl TryFrom<$name> for Vec<Gf32Bit> {
```

Comment:
is it better to define conversion to an array for each BA type? we know the size at compile time, so we can save some cost here

Reply:
Wouldn't each Array/GenericArray be its own type? When generating the vector/array of Gf32Bit from different BA types it would be nice to receive the same type. A function like 
```
fn<B:BooleanArray>(input: B)
{
    let b = Vec<Gf32Bit>::from(input);
}
```
works. When using an array, I think I would need to use a constant generic or arraylength something like this
```
fn<B:BooleanArray, U:ArrayLength>(input: B)
{
    let b = <GenericArray<Gf32Bit,U>>::from(input);
}
```

Reply:
For performance reason, I agree with you, we should definitely replace Vec. I first want to ship something that is functioning and will take it on later: https://github.com/private-attribution/ipa/issues/1254.